### DISCUSSION TITLE
How to setup energy prices

### DISCUSSION THREAD

**Discussion by Plato42** [jmcollin78/solar_optimizer#30] [Q&A]
Hello, I'm currently attempting to configure the Solar Optimizer, but I'm encountering an issue. It appears that the configuration doesn't permit fixed values (1), and it's also not accepting the dynamic electricity price value from my Tibber integration (sensor.electricity_price_casa).

I've thoroughly reviewed the instructions, but unfortunately, they don't provide any guidance on what steps to take next. Should I create "dummy" entities in Home Assistant first? If so, could you please advise on how these entities should be set up to ensure they're recognized by the Solar Optimizer? I apologize if I overlooked any details in the documentation.

![image](https://github.com/jmcollin78/solar_optimizer/assets/113471832/08c7e17a-d113-466c-b31c-5f9eccb0b42b)

**Comment by jmcollin78** [comment]
Hello,


You should create input_number with the cost values and put this input_number in the required field. Then you will be able to have dynamic prices. Sensor are not accepted because they can hold any kind of information (temperature, ...) and not prices.

**Reply by zefrenchtoon** [reply]
Hello,

To manage low and high prices from my energy vendor, I did a "numeric type" "Template" sensor to define the energy cost depending of the current time.
If I correctly understand your answer, this should not work ?
If so, how did you manage this please ? (Now, I have no access to teleinfo from HA so I have to manage it "manually" but I have an enphase system as you)

Also, FYI, the unit used for energy cost is never mentioned. I had to check your source code to see that you are looking for cents and not full currency. I think this should be mentioned. :) 


By the way, this seems to be a promising tool, big thanks to you. :)

**Reply by jmcollin78** [reply]
A numeric template should work if the entity is seen as a input_number.
I have input_number and a automation that set the current price at each hours change.
